### DISCUSSION TITLE
Keep showing unstaged file diff when setting focus to message pane.

### DISCUSSION THREAD

**Discussion by mcsilver** [gitextensions/gitextensions#11513] [Q&A]
I searched the forum for "commit focus" but came up empty. Not sure what else to search for, so...

When I am in the Commit window, I like to type in the message pane about the currently selected file's changes _before_ staging the file. That way, when I have several changes to write about, I know that the unstaged files are not yet included in the message while all the staged files are. However, whenever I click on the message pane to add text, the diff window's contents changes to the selected _staged_ file instead of remaining on the selected _unstaged_ file.

Is there already a setting that I can change, or is this a feature request? Or is there a way for me to mark the files that I've already written about without repurposing their unstaged/staged status?

(Yes, I know that a commit should be concise and not include multiple changes that each need writing about, but sometimes, that's just not the case.)

**Comment by vbjay** [comment]
The purpose is to review the files to be staged not the files not staged. It is on purpose to switch to the first staged file so you are looking at the diff of what is being committed. This allows quick keyboard commands to cycle through the what is going on commit and look at the diffs. 

Most people stage a change and write about it next. They may do this multiple times till ready to commit. Kind of why the staging up is left and message is right in dialog.

**Comment by mcsilver** [comment]
OK
